### ISSUE TITLE
Unable to add jvm option via admin UI /CUSTCOM-234

### ISSUE THREAD

**Issue by StaryiHome** [payara/Payara#4578]
# Description #
----------

When I try to add any JVM option, e.g. -Da=b, in the administration console section through Configurations-> server-config-> JVM Settings-> JVM Options-> Add JVM Option, I get the following error:

An error has occurred
JVM option -Xbootclasspath/a:${com.sun.aas.installRoot}/lib/grizzly-npn-api.jar already exists in the configuration.

In server.log I see

```
[2020-03-18T17:12:25.978+0700] [Payara 5.201] [WARNING] [] [] [tid: _ThreadID=200 _ThreadName=GF Domain XML] [timeMillis: 1584526345978] [levelValue: 900] [[
  Unprocessed event : UnprocessedChangeEvent{PropertyName=jvm-options, OldValue = [-server, [9|]--add-opens=java.base/jdk.internal.loader=ALL-UNNAMED, [9|]--add-opens=jdk.management/com.sun.management.internal=ALL-UNNAMED, [9|]--add-exports=java.base/jdk.internal.ref=ALL-UNNAMED, [9|]--add-opens=java.base/java.lang=ALL-UNNAMED, [9|]--add-opens=java.base/java.net=ALL-UNNAMED, [9|]--add-opens=java.base/java.nio=ALL-UNNAMED, [9|]--add-opens=java.base/java.util=ALL-UNNAMED, [9|]--add-opens=java.base/sun.nio.ch=ALL-UNNAMED, [9|]--add-opens=java.management/sun.management=ALL-UNNAMED, [9|]--add-opens=java.base/sun.net.www.protocol.jrt=ALL-UNNAMED, [9|]--add-opens=java.base/sun.net.www.protocol.jar=ALL-UNNAMED, [9|]--add-opens=java.naming/javax.naming.spi=ALL-UNNAMED, [9|]--add-opens=java.rmi/sun.rmi.transport=ALL-UNNAMED, [9|]--add-opens=java.logging/java.util.logging=ALL-UNNAMED, -Djava.awt.headless=true, -Djdk.corba.allowOutputStreamSubclass=true, -Djavax.xml.accessExternalSchema=all, -Djavax.management.builder.initial=com.sun.enterprise.v3.admin.AppServerMBeanServerBuilder, -XX:+UnlockDiagnosticVMOptions, -Djava.security.policy=${com.sun.aas.instanceRoot}/config/server.policy, -Djava.security.auth.login.config=${com.sun.aas.instanceRoot}/config/login.conf, -Dcom.sun.enterprise.security.httpsOutboundKeyAlias=s1as, -Xmx512m, -Djavax.net.ssl.keyStore=${com.sun.aas.instanceRoot}/config/keystore.jks, -Djavax.net.ssl.trustStore=${com.sun.aas.instanceRoot}/config/cacerts.jks, -Djdbc.drivers=org.h2.Driver, -DANTLR_USE_DIRECT_CLASS_LOADING=true, -Dcom.sun.enterprise.config.config_environment_factory_class=com.sun.enterprise.config.serverbeans.AppserverConfigEnvironmentFactory, -Dorg.glassfish.additionalOSGiBundlesToStart=org.apache.felix.shell,org.apache.felix.gogo.runtime,org.apache.felix.gogo.shell,org.apache.felix.gogo.command,org.apache.felix.shell.remote,org.apache.felix.fileinstall, -Dosgi.shell.telnet.port=6666, -Dosgi.shell.telnet.maxconn=1, -Dosgi.shell.telnet.ip=127.0.0.1, -Dgosh.args=--nointeractive, -Dfelix.fileinstall.dir=${com.sun.aas.installRoot}/modules/autostart/, -Dfelix.fileinstall.poll=5000, -Dfelix.fileinstall.log.level=2, -Dfelix.fileinstall.bundles.new.start=true, -Dfelix.fileinstall.bundles.startTransient=true, -Dfelix.fileinstall.disableConfigSave=false, -XX:NewRatio=2, -Dcom.ctc.wstx.returnNullForDefaultNamespace=true, -Djdk.tls.rejectClientInitiatedRenegotiation=true, -Dorg.jboss.weld.serialization.beanIdentifierIndexOptimization=false, -Dorg.glassfish.grizzly.DEFAULT_MEMORY_MANAGER=org.glassfish.grizzly.memory.HeapMemoryManager, -Dorg.glassfish.grizzly.nio.DefaultSelectorHandler.force-selector-spin-detection=true, [1.8.0|1.8.0u120]-Xbootclasspath/p:${com.sun.aas.installRoot}/lib/grizzly-npn-bootstrap-1.6.jar, [1.8.0u121|1.8.0u160]-Xbootclasspath/p:${com.sun.aas.installRoot}/lib/grizzly-npn-bootstrap-1.7.jar, [1.8.0u161|1.8.0u190]-Xbootclasspath/p:${com.sun.aas.installRoot}/lib/grizzly-npn-bootstrap-1.8.jar, [1.8.0u191|1.8.0u500]-Xbootclasspath/p:${com.sun.aas.installRoot}/lib/grizzly-npn-bootstrap-1.8.1.jar, [9|]-Xbootclasspath/a:${com.sun.aas.installRoot}/lib/grizzly-npn-api.jar, [|8]-Djava.endorsed.dirs=${com.sun.aas.installRoot}/modules/endorsed${path.separator}${com.sun.aas.installRoot}/lib/endorsed, [|8]-Djava.ext.dirs=${com.sun.aas.javaRoot}/lib/ext${path.separator}${com.sun.aas.javaRoot}/jre/lib/ext${path.separator}${com.sun.aas.instanceRoot}/lib/ext, [Azul-1.8.0u222|1.8.0u500]-XX:+UseOpenJSSE], NewValue = [[9|]--add-opens=java.base/jdk.internal.loader=ALL-UNNAMED, [9|]--add-opens=jdk.management/com.sun.management.internal=ALL-UNNAMED, [9|]--add-exports=java.base/jdk.internal.ref=ALL-UNNAMED, [9|]--add-opens=java.base/java.lang=ALL-UNNAMED, [9|]--add-opens=java.base/java.net=ALL-UNNAMED, [9|]--add-opens=java.base/java.nio=ALL-UNNAMED, [9|]--add-opens=java.base/java.util=ALL-UNNAMED, [9|]--add-opens=java.base/sun.nio.ch=ALL-UNNAMED, [9|]--add-opens=java.management/sun.management=ALL-UNNAMED, [9|]--add-opens=java.base/sun.net.www.protocol.jrt=ALL-UNNAMED, [9|]--add-opens=java.base/sun.net.www.protocol.jar=ALL-UNNAMED, [9|]--add-opens=java.naming/javax.naming.spi=ALL-UNNAMED, [9|]--add-opens=java.rmi/sun.rmi.transport=ALL-UNNAMED, [9|]--add-opens=java.logging/java.util.logging=ALL-UNNAMED, [1.8.0|1.8.0u120]-Xbootclasspath/p:${com.sun.aas.installRoot}/lib/grizzly-npn-bootstrap-1.6.jar, [1.8.0u121|1.8.0u160]-Xbootclasspath/p:${com.sun.aas.installRoot}/lib/grizzly-npn-bootstrap-1.7.jar, [1.8.0u161|1.8.0u190]-Xbootclasspath/p:${com.sun.aas.installRoot}/lib/grizzly-npn-bootstrap-1.8.jar, [1.8.0u191|1.8.0u500]-Xbootclasspath/p:${com.sun.aas.installRoot}/lib/grizzly-npn-bootstrap-1.8.1.jar, [9|]-Xbootclasspath/a:${com.sun.aas.installRoot}/lib/grizzly-npn-api.jar, [|8]-Djava.endorsed.dirs=${com.sun.aas.installRoot}/modules/endorsed${path.separator}${com.sun.aas.installRoot}/lib/endorsed, [|8]-Djava.ext.dirs=${com.sun.aas.javaRoot}/lib/ext${path.separator}${com.sun.aas.javaRoot}/jre/lib/ext${path.separator}${com.sun.aas.instanceRoot}/lib/ext, [Azul-1.8.0u222|1.8.0u500]-XX:+UseOpenJSSE], Source = GlassFishConfigBean.com.sun.enterprise.config.serverbeans.JavaConfig}, reason = Removal of: -Djava.security.policy=${com.sun.aas.instanceRoot}/config/server.policy cannot take effect without server restart, Removal of: -Djavax.xml.accessExternalSchema=all cannot take effect without server restart, Removal of: -Djava.awt.headless=true cannot take effect without server restart, Removal of: -Djavax.management.builder.initial=com.sun.enterprise.v3.admin.AppServerMBeanServerBuilder cannot take effect without server restart, Removal of: -Xmx512m cannot take effect without server restart, Removal of: -XX:NewRatio=2 cannot take effect without server restart, Removal of: -Djavax.net.ssl.trustStore=${com.sun.aas.instanceRoot}/config/cacerts.jks cannot take effect without server restart, Removal of: -server cannot take effect without server restart, Removal of: -Djavax.net.ssl.keyStore=${com.sun.aas.instanceRoot}/config/keystore.jks cannot take effect without server restart, Removal of: -XX:+UnlockDiagnosticVMOptions cannot take effect without server restart, Removal of: -Djava.security.auth.login.config=${com.sun.aas.instanceRoot}/config/login.conf cannot take effect without server restart, when = 1584526345977]]

[2020-03-18T17:12:26.005+0700] [Payara 5.201] [SEVERE] [] [org.glassfish.admingui] [tid: _ThreadID=258 _ThreadName=admin-thread-pool::admin-listener(5)] [timeMillis: 1584526346005] [levelValue: 1000] [[
  RestResponse.getResponse() gives FAILURE.  endpoint = 'http://localhost:4848/management/domain/configs/config/server-config/java-config/jvm-options'; attrs = '{-Djdk.corba.allowOutputStreamSubclass\=true, -Djavax.management.builder.initial\=com.sun.enterprise.v3.admin.AppServerMBeanServerBuilder, -Djdk.tls.rejectClientInitiatedRenegotiation\=true, -Dosgi.shell.telnet.maxconn\=1, -Dcom.sun.enterprise.config.config_environment_factory_class\=com.sun.enterprise.config.serverbeans.AppserverConfigEnvironmentFactory, -Da\=b, -Dgosh.args\=--nointeractive, [9|]-Xbootclasspath\/a\:$\{com.sun.aas.installRoot\}\/lib\/grizzly-npn-api.jar=, -server=, -Dorg.glassfish.additionalOSGiBundlesToStart\=org.apache.felix.shell,org.apache.felix.gogo.runtime,org.apache.felix.gogo.shell,org.apache.felix.gogo.command,org.apache.felix.shell.remote,org.apache.felix.fileinstall, -Djava.security.auth.login.config\=$\{com.sun.aas.instanceRoot\}\/config\/login.conf, -Dfelix.fileinstall.disableConfigSave\=false, -Djava.security.policy\=$\{com.sun.aas.instanceRoot\}\/config\/server.policy, -Dfelix.fileinstall.bundles.new.start\=true, -Dorg.glassfish.grizzly.DEFAULT_MEMORY_MANAGER\=org.glassfish.grizzly.memory.HeapMemoryManager, -Djavax.xml.accessExternalSchema\=all, -Dosgi.shell.telnet.port\=6666, -Dfelix.fileinstall.log.level\=2, [1.8.0.121|1.8.0.160]-Xbootclasspath\/p\:$\{com.sun.aas.installRoot\}\/lib\/grizzly-npn-bootstrap-1.7.jar=, [9|]--add-opens\=java.logging\/java.util.logging\=ALL-UNNAMED, [1.8.0.222|1.8.0.500]-XX\:+UseOpenJSSE=, -Djava.awt.headless\=true, -Dfelix.fileinstall.poll\=5000, [|8]-Djava.endorsed.dirs\=$\{com.sun.aas.installRoot\}\/modules\/endorsed$\{path.separator\}$\{com.sun.aas.installRoot\}\/lib\/endorsed, -Djdbc.drivers\=org.h2.Driver, -XX\:+UnlockDiagnosticVMOptions=, -Dfelix.fileinstall.dir\=$\{com.sun.aas.installRoot\}\/modules\/autostart\/, [1.8.0|1.8.0.120]-Xbootclasspath\/p\:$\{com.sun.aas.installRoot\}\/lib\/grizzly-npn-bootstrap-1.6.jar=, [1.8.0.161|1.8.0.190]-Xbootclasspath\/p\:$\{com.sun.aas.installRoot\}\/lib\/grizzly-npn-bootstrap-1.8.jar=, [1.8.0.191|1.8.0.500]-Xbootclasspath\/p\:$\{com.sun.aas.installRoot\}\/lib\/grizzly-npn-bootstrap-1.8.1.jar=, -Djavax.net.ssl.keyStore\=$\{com.sun.aas.instanceRoot\}\/config\/keystore.jks, -DANTLR_USE_DIRECT_CLASS_LOADING\=true, -Xmx512m=, profiler=false, -Dfelix.fileinstall.bundles.startTransient\=true, -Dorg.glassfish.grizzly.nio.DefaultSelectorHandler.force-selector-spin-detection\=true, target=server-config, -Dcom.ctc.wstx.returnNullForDefaultNamespace\=true, -Dosgi.shell.telnet.ip\=127.0.0.1, -Dcom.sun.enterprise.security.httpsOutboundKeyAlias\=s1as, -Djavax.net.ssl.trustStore\=$\{com.sun.aas.instanceRoot\}\/config\/cacerts.jks, -Dorg.jboss.weld.serialization.beanIdentifierIndexOptimization\=false, [9|]--add-exports\=java.base\/jdk.internal.ref\=ALL-UNNAMED, [|8]-Djava.ext.dirs\=$\{com.sun.aas.javaRoot\}\/lib\/ext$\{path.separator\}$\{com.sun.aas.javaRoot\}\/jre\/lib\/ext$\{path.separator\}$\{com.sun.aas.instanceRoot\}\/lib\/ext, -XX\:NewRatio\=2}']]
```
On the previous version 5.194, this error does not occur.

## Expected Outcome

JVM option successfully added.

## Current Outcome

JVM option add fail.

## Steps to reproduce (Only for bug reports) 

1. Installing Payara Server (according to README.txt)
2. Starting Payara Server (according to README.txt)
3. Go to web administration console: http://localhost:4848
4. Configurations-> server-config-> JVM Settings-> JVM Options-> Add JVM Option
5. Enter -Da=b as Value
6. Save

## Environment ##

- **Payara Version**: 5.201
- **Edition**: Full
- **JDK Version**: OpenJDK 1.8.0_242, Oracle JDK 1.8.0_181, Oracle JDK 1.8.0_201
- **Operating System**: LMDE 3 Cindy 4.9.0-12-amd64, Windows 10, Windows 7
- **Database**:

**Comment by fturizo** [comment]
Thanks for reporting this error, @StaryiHome. The defect is easily reproducible, so I have raised the internal JIRA issue `CUSTCOM-234` for our engineers to fix this ASAP.

**Comment by klementszabolcs** [comment]
In version 5.194 this error is missing. the problem is only from admin web page, with asadmin CLI isnt. I think the admin page is delete all jvm property first and add all, but the options with min/max version isnt success to remove  because the java version condition is apply with property remove.

**Comment by Chalfouh** [comment]
This still a problem in Payara Server 5.201 #badassfish (build 512)
i have the same issue

**Comment by FiruzzZ** [comment]
> In version 5.194 this error is missing. the problem is only from admin web page, with asadmin CLI isnt. I think the admin page is delete all JVM property first and add all, but the options with min/max version aren't successfully to remove because the java version condition is apply with property remove.

As you said, it removes all JVM and replaces it with a completely different set of JVM options and of course also ruins the .bak file too making impossible to recover them

**Comment by rpydpd** [comment]
This bug, or related functionality, is also particularly insidious because it has the potential to corrupt existing JVM parameters. 

In a `domain.xml` configuration file with the rather complex, but valid, `javaagent` argument (`-javaagent:/usr/payara5/jmx_prometheus_javaagent-0.11.0.jar=9556:/usr/payara5/prometheus_config.yaml`), if a new JVM argument is added via the UI, it will succeed. However, refreshing the JVM options page will reveal duplicate JVM option entries of existing parameters, but with equals signs appended, like '-Xmx:6G=' and possibly corrupted ones, like `-javaagent:/usr/payara5/jmx_prometheus_javaagent-0.11.0.jar=-X:WhateverParameter`. 

There is no way to delete these options via the UI, either. Instance restarts will fail due to the malformed JVM parameters.

The least invasive resolution is to utilize `asadmin --target XYZ delete-jvm-options`. If there are particularly complex JVM arguments, be sure to surround the string with single quotes and escape all colons with backslashes, i.e., `delete-jvm-options ''javaagent\:/usr/payara5/jmx_prometheus_javaagent-0.11.0.jar=-X\:CustomParameter'`.

**Comment by smillidge** [comment]
This is fixed in latest 5.2020.2
